tests/acosh_complex.cpp

```diff
@@ -1,21 +1,37 @@
 #include "test_helper.hpp"
 
-template <typename T> struct test_acosh {
-  bool operator()(sycl::queue &Q, T init_re, T init_im,
-                  bool is_error_checking = false) {
-    bool pass = true;
+TEMPLATE_TEST_CASE("Test complex acosh", "[acosh]", double, float, sycl::half) {
+  using T = TestType;
+  using std::make_tuple;
 
-    auto std_in = init_std_complex(init_re, init_im);
-    sycl::ext::cplx::complex<T> cplx_input{init_re, init_im};
+  sycl::queue Q;
+
+  cmplx<T> input;
+  bool is_error_checking;
+
+  std::tie(input, is_error_checking) = GENERATE(table<cmplx<T>, bool>(
+      {make_tuple(cmplx<T>{4.42, 2.02}, false),
+       make_tuple(cmplx<T>{inf_val<T>, 2.02}, true),
+       make_tuple(cmplx<T>{4.42, inf_val<T>}, true),
+       make_tuple(cmplx<T>{inf_val<T>, inf_val<T>}, true),
+       make_tuple(cmplx<T>{nan_val<T>, 2.02}, true),
+       make_tuple(cmplx<T>{4.42, nan_val<T>}, true),
+       make_tuple(cmplx<T>{nan_val<T>, nan_val<T>}, true),
+       make_tuple(cmplx<T>{nan_val<T>, inf_val<T>}, true),
+       make_tuple(cmplx<T>{inf_val<T>, nan_val<T>}, true)}));
+
+  auto std_in = init_std_complex(input);
+  sycl::ext::cplx::complex<T> cplx_input{input.re, input.im};
 
-    std::complex<T> std_out{init_re, init_im};
-    auto *cplx_out = sycl::malloc_shared<sycl::ext::cplx::complex<T>>(1, Q);
+  std::complex<T> std_out{input.re, input.im};
```

Comment:
I guess we can just do `std_out{input}`

Reply:
So unfortunately I cant do `std_out{input}` as input is of type `cmplx` which is just a helper struct of real and imaginary components. std::complex does not have a constructor for my helper struct `cmplx`. 

Also in this case we don't want the `half's` to be be translated into `floats` as this is the output value which should have the half type. As no operation is being performed on the output type this is valid, the reason we are initializing it to the input values is for the case were we are checking if `acosh` can undo `cosh`. However when error codes are tested this value is overwritten.